### PR TITLE
moved business logic of workforce page into the specflow file

### PR DESCRIPTION
**Pull request by Faizi52996**
moved the business logic into specflow.
[AB#185925](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/185925)

### CODE REVIEW

**Review by jrabbott — CHANGES_REQUESTED** [review]

File: web/tests/EducationBenchmarking.Web.E2ETests/Steps/BenchmarkWorkforceSteps.cs [hunk 1]
```diff
@@ -85,4 +85,19 @@ public async Task ThenChartViewIsShowingOnWorkforcePage()
...
     }
+    
+    [When("I click the dimension dropdown for '(.*)'")]
+    public async Task WhenIClickTheDimensionDropdownFor(string chartName)
+    {
+        await _benchmarkWorkforcePage.ClickOnDimensionDropdown(chartName);
+    }
+    
+    [Then("the '(.*)' are showing in the dimension dropdown for '(.*)'")]
+    public async Task ThenTheAreShowingInTheDimensionDropdownFor(string dropdownOptions, string chartName)
```
[the remark this answers is not in the document]
Comment (Faizi52996) [reply web/tests/EducationBenchmarking.Web.E2ETests/Steps/BenchmarkWorkforceSteps.cs]: this is interesting and done.

File: web/tests/EducationBenchmarking.Web.E2ETests/Steps/BenchmarkWorkforceSteps.cs [hunk 2]
```diff
@@ -85,4 +85,23 @@ public async Task ThenChartViewIsShowingOnWorkforcePage()
...
+    {
+        await _benchmarkWorkforcePage.ClickOnDimensionDropdown(chartName);
+    }
+    
+    [Then("the '(.*)' are showing in the dimension dropdown for '(.*)'")]
+    public async Task ThenTheAreShowingInTheDimensionDropdownFor(string[] dropdownOptions, string chartName)
+    {
+        var chartDimensionLocator = _benchmarkWorkforcePage.GetChartDimensionDropdown(chartName);
+       await _benchmarkWorkforcePage.AssertDropDownDimensions(chartDimensionLocator, dropdownOptions);
+    }
```
Comment (jrabbott) [inline web/tests/EducationBenchmarking.Web.E2ETests/Steps/BenchmarkWorkforceSteps.cs:100]: Spacing

File: web/tests/EducationBenchmarking.Web.E2ETests/Steps/BenchmarkWorkforceSteps.cs [hunk 3]
```diff
@@ -85,4 +85,23 @@ public async Task ThenChartViewIsShowingOnWorkforcePage()
...
+    [StepArgumentTransformation]
+    public string[] TransformTDropdownOptionsToListOfStrings(string commaSeperatedList)
+    {
+        return commaSeperatedList.Split(',').Select(option => option.Trim()).ToArray();
+    }
+
```
Comment (jrabbott) [inline web/tests/EducationBenchmarking.Web.E2ETests/Steps/BenchmarkWorkforceSteps.cs:106]: Spacing

File: web/tests/EducationBenchmarking.Web.E2ETests/Pages/BenchmarkWorkforcePage.cs
Comment (jrabbott) [inline web/tests/EducationBenchmarking.Web.E2ETests/Pages/BenchmarkWorkforcePage.cs:1]: Unused const "DefaultDimensionOption" needs removing

**Review by jrabbott — CHANGES_REQUESTED** [review]

**Review by jrabbott — APPROVED** [review]

**Pull request merged**